src/scope-analyzer.js

```diff
@@ -248,20 +260,20 @@ export default class ScopeAnalyzer extends MonoidalReducer {
   reduceSwitchStatement(node, { discriminant, cases }) {
     return this
       .fold(cases)
+      .withPotentialVarFunctions(getFunctionDeclarations([].concat(...node.cases.map(c => c.consequent))))
       .finish(node, ScopeType.BLOCK)
-      .withPotentialVarFunctions(getFunctionDeclarations([...node.cases.map(c => c.consequent)]))
       .concat(discriminant);
   }
 
   reduceSwitchStatementWithDefault(node, { discriminant, preDefaultCases, defaultCase, postDefaultCases }) {
-    const functionDeclarations = getFunctionDeclarations([
+    const functionDeclarations = getFunctionDeclarations([].concat(
```

Comment:
I don't understand why we need to do flattening now but didn't before.

Reply:
We did need to; this was just broken. The only test for cases had blocks around each case, so the broken behavior wasn't exercised.

Reply:
Oh god.